### PR TITLE
(fleet) refactor RC bits in the updater

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

- Decouple RC boilerplate from the business logic in the updater code
- Added tests for all the RC boilerplate
- Also bundled some minor fixes (unwanted logs, obtaining the state failing when there was no stable, errors when the lock repository did not exist)

### Motivation

RC's boilerplate was making everything around it hard to understand.

### CODE REVIEW

**Review by raphaelgavache — APPROVED** [review]

File: pkg/updater/download.go [hunk 1]
```diff
@@ -198,6 +198,8 @@ func extractTarArchive(archivePath string, destinationPath string, compression c
 			if err != nil {
 				return fmt.Errorf("could not create symlink: %w", err)
 			}
+		case tar.TypeLink:
+			// we currently don't support hard links in the updater
```
Comment (BaptisteFoy) [inline pkg/updater/download.go:202]: **👏 Praise:** ‏Nice. Should we add a todo ticket?

File: pkg/updater/remote_config.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	Arch     string `json:"arch"`
+}
+
+type catalog struct {
+	Packages []Package `json:"packages"`
+}
+
+func (c *catalog) getPackage(pkg string, version string, arch string, platform string) (Package, bool) {
+	for _, p := range c.Packages {
+		if p.Name == pkg && p.Version == version && (p.Arch == "" || p.Arch == arch) && (p.Platform == "" || p.Platform == platform) {
```
Comment (BaptisteFoy) [inline pkg/updater/remote_config.go:76]: **💭 Thought:** ‏Kinda wondering if we should support empty archs & platforms in the future

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/updater/download.go (hunk 1 above)
Comment (arbll) [reply pkg/updater/download.go:202]: We can but I would not pro-actively work on this until it blocks something given the added complexity

File: pkg/updater/remote_config.go (hunk 2 above)
Comment (arbll) [reply pkg/updater/remote_config.go:76]: Had the same thought and I concluded that we might have packages that are cross-arch compatible (ex: some tracers)

**Pull request merged**